extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/DbCheckingPMetaSupplier.java

```diff
@@ -0,0 +1,160 @@
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.datamodel.Tuple2;
+import com.hazelcast.jet.impl.util.Util;
+import com.hazelcast.jet.mongodb.dataconnection.MongoDataConnection;
+import com.hazelcast.jet.pipeline.DataConnectionRef;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoDatabase;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.security.Permission;
+import java.util.List;
+import java.util.function.Function;
+
+import static com.hazelcast.jet.datamodel.Tuple2.tuple2;
+import static com.hazelcast.jet.mongodb.impl.MongoUtilities.checkCollectionExists;
+import static com.hazelcast.jet.mongodb.impl.MongoUtilities.checkDatabaseExists;
+
+/**
+ * A {@link ProcessorMetaSupplier} that will check if requested database and collection exist before creating
+ * the processors.
+ */
+public class DbCheckingPMetaSupplier implements ProcessorMetaSupplier {
+
+    private final Permission requiredPermission;
+    private final boolean shouldCheckOnEachCall;
+    private final ProcessorMetaSupplier standardForceOnePMS;
+    private final boolean forceTotalParallelismOne;
+    private final String databaseName;
+    private final String collectionName;
+    private final ProcessorSupplier processorSupplier;
+    private final SupplierEx<? extends MongoClient> clientSupplier;
+    private final DataConnectionRef dataConnectionRef;
+    private final int preferredLocalParallelism;
+
+    /**
+     * Creates a new instance of this meta supplier.
+     */
+    public DbCheckingPMetaSupplier(@Nullable Permission requiredPermission,
+                                   boolean shouldCheckOnEachCall,
+                                   boolean forceTotalParallelismOne,
+                                   @Nullable String databaseName,
+                                   @Nullable String collectionName,
+                                   @Nullable SupplierEx<? extends MongoClient> clientSupplier,
+                                   @Nullable DataConnectionRef dataConnectionRef,
+                                   @Nonnull ProcessorSupplier processorSupplier,
+                                   int preferredLocalParallelism
+    ) {
+        this.requiredPermission = requiredPermission;
+        this.shouldCheckOnEachCall = shouldCheckOnEachCall;
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        this.databaseName = databaseName;
+        this.collectionName = collectionName;
+        this.processorSupplier = processorSupplier;
+        this.clientSupplier = clientSupplier;
+        this.dataConnectionRef = dataConnectionRef;
+        this.preferredLocalParallelism = forceTotalParallelismOne ? 1 : preferredLocalParallelism;
+        this.standardForceOnePMS = ProcessorMetaSupplier.forceTotalParallelismOne(processorSupplier);
```

Comment:
is there any reason to use random member rather than local member in case of `standardForceOnePMS`?

Reply:
7831956e3b2ce0787afc566a54b89fa469b724b0

Reply:
ok, was it impossibility to get local address in constructor?

Reply:
I leave it up to you: we can keep it as it is, run on random member or add new variant of `forceTotalParallelismOne` which determines local member in `PMS.init` so the `DbCheckingPMetaSupplier` could be immutable